src/components/Trade/TradeInput.tsx

```diff
@@ -428,13 +415,34 @@ export const TradeInput = () => {
     [sellAmountCryptoPrecision, buyAmountCryptoPrecision, isTradeQuotePending],
   )
 
+  const hasSufficientProtocolFeeBalances = useMemo(() => {
+    if (protocolFees === undefined) return true
```

Comment:
would false be safer here as a default, and check against explicitly 0, or is this undefined for things like cowswap?

Reply:
they should always be defined so agree that undefined object means something is broken and we should return false